packages/react-reconciler/src/ReactFiberNewContext.new.js

```diff
@@ -141,9 +141,11 @@ export function popProvider(
 export function scheduleWorkOnParentPath(
   parent: Fiber | null,
   renderLanes: Lanes,
+  stopAt: Fiber | null,
 ) {
   // Update the child lanes of all the ancestors, including the alternates.
   let node = parent;
+  const stopAtAlternate = stopAt !== null ? stopAt.alternate : null;
```

Comment:
You shouldn't need to check the alternate because we set `.return` on the parent before continuing on a child. If this wasn't true, there would be worse bugs because it would mean we're traversing the wrong tree. So I would replace this with a dev-only warning.